### ISSUE TITLE
Datatable: multi-sort loses order with lazy

### ISSUE THREAD

**Issue by hecnabae** [primefaces/primefaces#6468]
**Describe the defect**
I've a table with columns [a,b,c,d,e] and lazy initialization. When I provide an intilial list of sortmeta (c, a) it sorts data correctly. But, If I change page, It loses sortmeta order (a, c). For example:

After apply sort columns:
- sortMeta: c DESC, a DESC

After change page (a and c got swapped):
- sortMeta: a DESC, c DESC 

**Environment:**
 - PF Version: 8.0
 - JSF 2.3: Jakarta faces 2.3.14
 - Affected browsers: ALL

**Expected behavior**
Maintain sortmeta order with pagination. 

**Example XHTML**
```html
<p:dataTable scrollable="true" scrollHeight="auto" rows="#{exampleBean.session.nReg}" first="#{exampleBean.session.nPage}" id="dtExample" paginator="true" paginatorTemplate="{CurrentPageReport} {FirstPageLink} {PreviousPageLink} {PageLinks} {NextPageLink} {LastPageLink} {RowsPerPageDropdown}" rowsPerPageTemplate="20,50,100" emptyMessage="Empty table" value="#{exampleBean.dataSource}" var="rec" paginatorAlwaysVisible="true" currentPageReportTemplate="{totalRecords}" selection="#{exampleBean.selectedRecords}" rowKey="#{rec.uid}" sortOrder="descending" style="font-size: 0.9em !important;" lazy="true" sortMode="multiple" styleClass="myTooltip" rowIndexVar="rowIndex" rowStyleClass="#{exampleBean.styleClazz(rec)">
        <p:column selectionMode="multiple" style="width:16px;text-align:center" exportable="false"/>
	<p:column headerText="A" sortBy="#{rec.a}">#{rec.a}</p:column>
        <p:column headerText="B" sortBy="#{rec.b}">#{rec.b}</p:column>
        <p:column headerText="C" sortBy="#{rec.c}">#{rec.c}</p:column>
        <p:column headerText="D" sortBy="#{rec.d}">#{rec.d}</p:column>
	<p:column headerText="E" sortBy="#{rec.e}">#{rec.e}</p:column>
</p:dataTable>
```

**Comment by tandraschko** [comment]
@Rapster could you check it and maybe fix it within your PR?

**Comment by Rapster** [comment]
Could you make a quick reproducer using https://github.com/primefaces/primefaces-test please? As I don't see your sortMeta input

**Comment by Rapster** [comment]
Fixed with #6390
